rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/TargetRepositoryEndpoint.java

```diff
@@ -0,0 +1,192 @@
+
+    static final String GET_SPECIFIC_DESC = "Gets a specific target repository.";
+
+    /**
+     * {@value GET_SPECIFIC_DESC}
+     *
+     * @param id {@value SCM_ID}
+     * @return
+     */
+    @Operation(
+            summary = GET_SPECIFIC_DESC,
+            responses = {
+                    @ApiResponse(
+                            responseCode = SUCCESS_CODE,
+                            description = SUCCESS_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = TargetRepository.class))),
+                    @ApiResponse(responseCode = NOT_FOUND_CODE, description = NOT_FOUND_DESCRIPTION),
+                    @ApiResponse(
+                            responseCode = INVALID_CODE,
+                            description = INVALID_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @GET
+    @Path("/{id}")
+    @Consumes(MediaType.APPLICATION_JSON_PATCH_JSON) // workaround for PATCH support
+    TargetRepository getSpecific(@Parameter(description = REPO_ID) @PathParam("id") String id);
+
+    static final String CREATE_NEW_DESC = "Creates a new target repository.";
+
+    /**
+     * {@value CREATE_NEW_DESC} {@value CREATE_NEW_DESC2}
+     *
+     * @param request
+     * @return
+     */
+    @Operation(
+            summary = CREATE_NEW_DESC,
+            responses = {
+                    @ApiResponse(
+                            responseCode = ENTITY_CREATED_CODE,
+                            description = ENTITY_CREATED_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = RepositoryCreationResponse.class))),
+                    @ApiResponse(
+                            responseCode = INVALID_CODE,
+                            description = INVALID_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = CONFLICTED_CODE,
+                            description = CONFLICTED_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @POST
+    @RespondWithStatus(Response.Status.CREATED)
+    TargetRepository createNew(@NotNull TargetRepository request);
```

Comment:
Shouldn't this endpoint be limited to admins?

Reply:
Yes, it definitely should. I forgot to limit it. Thank you!